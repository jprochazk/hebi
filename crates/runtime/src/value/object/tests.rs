use super::*;

#[test]
fn test_object() {
  let mut v = List::from([Value::int(0), Value::int(1), Value::int(2)]);
  v.push(Value::int(3));
  insta::assert_snapshot!(format!("display:\n{v}"));

  let mut v = Dict::from_iter([(Key::from("a"), Value::int(0))]);
  v.insert("b", Value::int(1));
  insta::assert_snapshot!(format!("display:\n{v}"));
}

#[test]
fn test_object_ptr() {
  let mut v = Handle::alloc(List::from([Value::int(0), Value::int(1), Value::int(2)]));
  v.push(Value::int(3));
  insta::assert_snapshot!(format!("display:\n{v}"));

  let mut v = Handle::alloc(Dict::from_iter([(Key::from("a"), Value::int(0))]));
  v.insert("b", Value::int(1));
  insta::assert_snapshot!(format!("display:\n{v}"));
}

#[test]
fn test_object_value() {
  let v = Value::object(Handle::alloc(List::from([
    Value::int(0),
    Value::int(1),
    Value::int(2),
  ])));
  v.clone().to_list().unwrap().push(Value::int(3));
  insta::assert_snapshot!(format!("display:\n{v}"));

  let v = Value::object(Handle::alloc(Dict::from_iter([(
    Key::from("a"),
    Value::int(0),
  )])));
  v.clone().to_dict().unwrap().insert("b", Value::int(1));
  insta::assert_snapshot!(format!("display:\n{v}"));
}
