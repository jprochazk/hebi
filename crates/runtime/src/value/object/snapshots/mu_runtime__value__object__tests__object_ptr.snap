---
source: crates/runtime/src/value/object/tests.rs
expression: "format!(\"display:\\n{v}\\n\\ndebug:\\n{v:#?}\")"
---
display:
[0, 1, 2, 3]

debug:
List(
    [
        Value {
            type: "int",
            value: 0,
        },
        Value {
            type: "int",
            value: 1,
        },
        Value {
            type: "int",
            value: 2,
        },
        Value {
            type: "int",
            value: 3,
        },
    ],
)
